### PR TITLE
feat(linter): no disabled tests(eslint-jest-plugin)

### PR DESCRIPTION
**Pull request by mysteryven**
I mark it as ready because I want to try other rules. I leave this case uncompleted:

```js
 ("import { test } from '@jest/globals';test('something');", None),
```

Will continue working on it when `SymbolId` is added to our AST.

### CODE REVIEW

File: crates/oxc_linter/src/jest_ast_util.rs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+fn resolve_to_jest_fn<'a>(
+    call_expr: &'a CallExpression,
+    ctx: &'a LintContext,
+) -> Option<ResolvedJestFn<'a>> {
+    let ident = resolve_first_ident(&call_expr.callee);
+
+    if let Some(ident) = ident {
+        if ctx.semantic().is_reference_to_global_variable(ident) {
+            return Some(ResolvedJestFn { local: Cow::Borrowed(ident.name.as_str()) });
```
Comment (mysteryven) [inline crates/oxc_linter/src/jest_ast_util.rs:106]: @Boshen I want to detect whether an identifier is a local variable or an import binding. Could you please give me some advice in this context?

My final aim is:
```js
//  should pass, the `test` is a customized method
import { test } from './test-utils'; test('something');

// should fail. 
import { test } from '@jest/globals'; test('something'); 
```
It related to [`resolveScope`](https://github.com/jest-community/eslint-plugin-jest/blob/a058f22f94774eeea7980ea2d1f24c6808bf3e2c/src/rules/utils/parseJestFnCall.ts#L558) in eslint-plugin-jest.

Comment (Boshen) [reply crates/oxc_linter/src/jest_ast_util.rs:106]: I just added this yesterday! https://github.com/Boshen/oxc/commit/c5402c1cae258d57793d4db5df3394134a7a99c5#diff-6a8fc122c14fff86e6a0b193287697a9b685de2b106f9413c6ed9d3c29c577be

File: _typos.toml
Comment (Boshen) [inline _typos.toml:1]: We have .typos.toml already, it's not working?

File: crates/oxc_linter/src/rules.rs [hunk 2]
```diff
@@ -57,7 +57,8 @@ oxc_macros::declare_all_lint_rules! {
     typescript::no_empty_interface,
     typescript::no_extra_non_null_assertion,
     typescript::no_non_null_asserted_optional_chain,
-    typescript::no_unnecessary_type_constraint
+    typescript::no_unnecessary_type_constraint,
+    eslint_plugin_jest::no_disabled_tests,
```
Comment (Boshen) [inline crates/oxc_linter/src/rules.rs:61]: Shall we keep it simple as `jest::no_disabled_tests`?

File: _typos.toml
Comment (mysteryven) [reply _typos.toml:1]: I missed it 😢

File: crates/oxc_linter/src/jest_ast_util.rs (hunk 1 above)
Comment (mysteryven) [reply crates/oxc_linter/src/jest_ast_util.rs:106]: Great Work! But I still didn't figure out. Is there some way to can get the `symbol_id` in this function?

File: crates/oxc_linter/src/rules/jest/no_disabled_tests.rs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+struct NoDisabledTestsDiagnostic(&'static str, &'static str, #[label] pub Span);
+
+const MESSAGES: Map<&'static str, (&'static str, &'static str)> = phf_map! {
+    "missingFunction" => ("Test is missing function argument", "Add function argument"),
+    "pending" => ("Call to pending()", "Remove pending() call"),
+    "disabledSuiteWithSkip" => ("Disabled test suite", "Remove the appending `.skip`"),
+    "disabledSuiteWithX" => ("Disabled test suite", "Remove x prefix"),
+    "disabledTestWithSkip" => ("Disabled test", "Remove the appending `.skip`"),
+    "disabledTestWithX" => ("Disabled test", "Remove x prefix"),
+};
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/jest/no_disabled_tests.rs:69]: This looks like an enum 🤔

File: crates/oxc_linter/src/jest_ast_util.rs [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+struct ResolvedJestFn<'a> {
+    pub local: Cow<'a, str>,
+}
+
+fn resolve_to_jest_fn<'a>(
+    call_expr: &'a CallExpression,
+    ctx: &'a LintContext,
+) -> Option<ResolvedJestFn<'a>> {
+    let ident = resolve_first_ident(&call_expr.callee);
```
Comment (Boshen) [inline crates/oxc_linter/src/jest_ast_util.rs:102]: I think you can add a `?` to the end and let this return early

File: crates/oxc_linter/src/rules/jest/no_disabled_tests.rs [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    ///
+    /// it('bar');
+    /// test('bar');
+    ///
+    /// it('foo', () => {
+    ///   pending();
+    /// });
+    /// ```
+    NoDisabledTests,
+    suspicious
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/jest/no_disabled_tests.rs:54]: Shall we put this under nursery, and add some todo comments for future references?

File: crates/oxc_linter/src/jest_ast_util.rs [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+    pub raw: Cow<'a, str>,
+}
+
+pub fn parse_jest_fn_call<'a>(
+    call_expr: &'a CallExpression,
+    ctx: &'a LintContext,
+) -> Option<ParsedJestFnCall<'a>> {
+    let callee = &call_expr.callee;
+
+    let resolved = resolve_to_jest_fn(call_expr, ctx);
```
Comment (Boshen) [inline crates/oxc_linter/src/jest_ast_util.rs:41]: nit: use `?` for returning early.

File: crates/oxc_linter/src/jest_ast_util.rs [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+    // if bailed out, we're not a jest function
+    resolved.as_ref()?;
```
Comment (Boshen) [inline crates/oxc_linter/src/jest_ast_util.rs:44]: 🤔 why is this on its own line

File: crates/oxc_linter/src/jest_ast_util.rs [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+        }
+
+        return Some(ParsedJestFnCall { kind, members, raw: first });
+    }
+
+    None
+}
+
+struct ResolvedJestFn<'a> {
+    pub local: Cow<'a, str>,
```
Comment (Boshen) [inline crates/oxc_linter/src/jest_ast_util.rs:95]: why do we need a cow? I don't see anything that's being owned. If everything comes from an Atom, you can also use `&'a Atom`

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/jest/no_disabled_tests.rs [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    NoDisabledTests,
+    nursery
+);
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(jest/no-disabled-tests): {0:?}")]
+#[diagnostic(severity(warning), help("{1:?}"))]
+struct NoDisabledTestsDiagnostic(&'static str, &'static str, #[label] pub Span);
+
+pub enum Message {
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/jest/no_disabled_tests.rs:61]: Some of the `pub`s seemed unnecessary.

**Comment by Boshen** [comment]
Anything left that blocks merging?

**Comment by mysteryven** [comment]
> Anything left that blocks merging?

Nothing, It's ready now.

File: crates/oxc_linter/src/rules/jest/no_disabled_tests.rs (hunk 9 above)
Comment (mysteryven) [reply crates/oxc_linter/src/rules/jest/no_disabled_tests.rs]: done.

**Review by Boshen — DISMISSED** [review]

**Comment by Boshen** [comment]
Fix the conflict and you can merge it yourself ;-)

**Review by Boshen — APPROVED** [review]

**Pull request merged**